### ISSUE TITLE
[FEAT] Imviz: Hint on setting up a custom color map (for docs)

### ISSUE THREAD

**Issue by hcferguson** [spacetelescope/jdaviz#1434]
### Description

For inspecting segmaps, it is very useful to have a colormap that does *not* have adjacent flux values have similar colors. Photutils provides a handy routine for making this colormap. With the current documentation, the incantation for using a custom colormap is a bit obscure, so it might help to provide something along the lines of the following hint in the Imviz docs.

> Sometimes it is useful to have a colormap that has contrasting colors for adjacent flux values. A common use case is inspecting segmentation maps. Photutils provides a routine to make a random colormap, which can be used for Imviz, following the example below (which assumes you are adjusting the second layer in the stack of images -- i.e. `layer[1]`:

```
from photutils.utils import make_random_cmap
randcmap = make_random_cmap(ncolors=256)
randcmap.colors[0] = 0. # To make your background black
viewer.state.layers[1].cmap = randcmap
```

### Additional context

Segmentations maps are integer images where each integer is the source ID, or zero if the pixel is identified as background. These are a standard output data product from from SExtractor or Photutils.  For inspecting segmentation maps it is particularly useful to have a colormap where adjacent entries have contrasting colors.

**Comment by pllim** [comment]
I also added an example of injecting your own colormap into Imviz over here:

* https://github.com/spacetelescope/jdaviz/blob/main/notebooks/concepts/imviz_custom_colormap.ipynb

@larrybradley was going to test it out but I have not heard back.

**Comment by larrybradley** [comment]
@hcferguson - The `SegmenationMap` instance also has a default (randomly generated) `cmap` attribute and a `make_cmap` method where the background color and random seed can be specified.

- https://photutils.readthedocs.io/en/latest/api/photutils.segmentation.SegmentationImage.html#photutils.segmentation.SegmentationImage.cmap

- https://photutils.readthedocs.io/en/latest/api/photutils.segmentation.SegmentationImage.html#photutils.segmentation.SegmentationImage.make_cmap


Here is my very simple example test notebook from last night:
https://gist.github.com/larrybradley/9c2629946b599874f9b52a7b757db16e

**Comment by pllim** [comment]
I don't think it is right to let user overwrite an existing colormap name with a completely different colormap. If user wants to do that, fine, but we should not recommend that in the documentation. I already have that concept notebook, so I am not sure what the ask is here. What else do you want so we can close this out? Thanks.

**Comment by hcferguson** [comment]
I don't think in my example I'm overwriting a colormap name.  I'm just remapping the colors in that layer without assigning a name. That example could be tweaked to use the `make_cmap` option that Larry recommends. 

I'm happy to close this one as long as we keep #1417 open. We could add an example of setting a custom colormap into that table of examples.

**Comment by pllim** [comment]
I thought @larrybradley said it couldn't get the API to update the colormap drop-down on Slack? But I am confused now looking at https://github.com/spacetelescope/jdaviz/issues/1434#issuecomment-1171339079

**Comment by larrybradley** [comment]
@pllim That was using the GUI.

**Comment by pllim** [comment]
Looking at Larry's gist, it is weird Astrowidgets is not recognizing that custom colormap as option even though it was added before Imviz as initialized because Astrowidgets polls Glue global state in real time. I'll have to investigare.

https://github.com/spacetelescope/jdaviz/blob/ff6e66886c85610320c8e846cd3cc2527d46a2a5/jdaviz/core/astrowidgets_api.py#L223

**Comment by larrybradley** [comment]
@pllim My custom colormap name did appear in the GUI dropdown after I did `colormaps.add('segm_cmap', cmap)`

**Comment by pllim** [comment]
Okay, so drop-down shows it but you cannot use `viewer.set_colormap('segm_cmap')`, is that correct? I am trying piece all these clues together.

**Comment by larrybradley** [comment]
Yes, that's correct.  Try my notebook.

**Comment by pllim** [comment]
Turns out what Larry found was a bug. Proposed fix and new documentation at #1440 .
